src/helpers/is-beta-feature.js

```diff
@@ -1,20 +1,25 @@
 'use strict'
 
+let navGroupCache = null
+let componentCache = null
+
 module.exports = (navUrl, { data: { root } }) => {
   const { contentCatalog, page } = root
-  if (page.layout === '404') return
-  if (contentCatalog === undefined) return
-  var pages = contentCatalog.navGroup
-  var cp = contentCatalog.cp
-  if (!pages || cp !== page.component.name) {
-    pages = contentCatalog.findBy({ component: page.component.name, family: 'page' })
-    contentCatalog.navGroup = pages
-    contentCatalog.cp = page.component.name
+  if (page.layout === '404') return false
+  if (!contentCatalog) return false
+
+  // Only perform lookup if caches are invalid or stale
+  if (!navGroupCache || componentCache !== page.component.name) {
+    navGroupCache = contentCatalog.findBy({ component: page.component.name, family: 'page' })
+    componentCache = page.component.name
   }
-  for (let i = 0; i < pages.length; i++) {
-    if (pages[i].pub.url === navUrl &&
-      pages[i].asciidoc.attributes['page-beta']) {
+
+  // Iterate through cached pages and check for beta status
+  for (let i = 0; i < navGroupCache.length; i++) {
+    if (navGroupCache[i].pub.url === navUrl && navGroupCache[i].asciidoc.attributes['page-beta']) {
       return true
```

Comment:
using foreach avoids accessing the same array multiple times
```suggestion
for (const navGroup of navGroupCache) {
    if (navGroup.pub.url === navUrl && navGroup.asciidoc.attributes['page-beta']) {
        return true;
    }
}
```